services/consensuscontext/validate_block_test.go

```diff
@@ -0,0 +1,130 @@
+package consensuscontext
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/config"
+	"github.com/orbs-network/orbs-network-go/crypto/digest"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-network-go/instrumentation/metric"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/stretchr/testify/require"
+	"os"
+	"testing"
+)
+
+// func (s *service) ValidateTransactionsBlock(ctx context.Context, input *services.ValidateTransactionsBlockInput) (*services.ValidateTransactionsBlockOutput, error) {
+func TestValidateTransactionBlock(t *testing.T) {
```

Comment:
I'd move all validation tests to unit tests without mocks, so that you don't have to perform all of this excessive stubbing. Take a look at the validation logic + test in txpool

Reply:
Will be part of a different PR, in branch feature/refactor-block-validation